src/PaymentGateways/Gateways/Stripe/Actions/CreatePaymentIntent.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe\Actions;
+
+use Give\LegacyPaymentGateways\Adapters\LegacyDonationData;
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\DonationSummary;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentIntent;
+use Give\PaymentGateways\Gateways\Stripe\ValueObjects\PaymentMethod;
+use Give\PaymentGateways\Gateways\Stripe\WorkflowAction;
+use Give\ValueObjects\Money;
+
+class CreatePaymentIntent extends WorkflowAction
+{
+    public function __invoke(
+        GatewayPaymentData $paymentData,
+        DonationSummary $donationSummary,
+        \Give_Stripe_Customer $giveStripeCustomer,
+        PaymentMethod $paymentMethod
+    )
+    {
+        /**
+         * This filter hook is used to update the payment intent arguments.
+         *
+         * @since 2.5.0
+         */
+        $intent_args = apply_filters(
+            'give_stripe_create_intent_args',
+            [
+                'amount' => Money::of($paymentData->price, $paymentData->currency)->getMinorAmount(),
+                'currency' => $paymentData->currency,
+                'payment_method_types' => ['card'],
+                'statement_descriptor' => give_stripe_get_statement_descriptor(),
+                'description' => $donationSummary,
+                'metadata' => give_stripe_prepare_metadata($paymentData->donationId, new LegacyDonationData( $paymentData, $paymentMethod->id() ) ),
+                'customer' => $giveStripeCustomer->get_id(),
+                'payment_method' => $paymentMethod->id(),
+                'confirm' => true,
+                'return_url' => $paymentData->redirectUrl,
+            ]
+        );
+
+        // Send Stripe Receipt emails when enabled.
+        if ( give_is_setting_enabled( give_get_option( 'stripe_receipt_emails' ) ) ) {
+            $intent_args['receipt_email'] = $paymentData->donorInfo->email;
+        }
+
+        $intent = give( PaymentIntent::class )->create( $intent_args );
+
+        give_insert_payment_note( $paymentData->donationId, 'Stripe Charge/Payment Intent ID: ' . $intent->id() );
```

Comment:
@kjohnson do you think we can internationalize these strings?

Reply:
Sure. A lot of these weren't previously translatable and were brought over via copy/paste. I can update to add translation.

Reply:
@ravinderk resolved by 6d29d4331.